meta-lmp-bsp/dynamic-layers/xilinx-tools/recipes-bsp/bitstream/bitstream-extraction_git.bb

```diff
@@ -0,0 +1,30 @@
+DESCRIPTION = "Recipe to extract bitstream"
+
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+
+DEPENDS += "virtual/hdf"
+
+PROVIDES = "virtual/bitstream"
+
+PACKAGE_ARCH ?= "${MACHINE_ARCH}"
+
+inherit xsctbit
+
+XSCTH_MISC = "-hwpname ${XSCTH_PROJ}_hwproj -hdf_type ${HDF_EXT}"
+
+do_compile[noexec] = "1"
+
+SYSROOT_DIRS += "${nonarch_base_libdir}/firmware"
+
+do_install() {
+
+    if [ -e ${XSCTH_WS}/${XSCTH_PROJ}_hwproj/*.bit ]; then
+        install -d ${D}/${nonarch_base_libdir}/firmware/
+        install -d ${D}/var/lib/firmware/bitstream/
```

Comment:
Var is a special directory in our setup because it is RW, so not really managed by ostree.

As all you need is a link, it is better to create a systemd tmpfile for this recipe, like https://github.com/foundriesio/meta-lmp/blob/master/meta-lmp-base/recipes-security/softhsm/softhsm/tmpfiles.conf (but using link instead, check https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html for more info). This allows the link to be created by systemd during boot automatically, and it will keep working even if the user erases /var completely (as it will be restored on a following boot).

Reply:
Actually, your recipe will have to install the link and the systemd tmpfiles configuration will have to create the directory (under var).

Reply:
the kernel driver will only load from /lib/firmware/... so the symlink has to point to someplace that is writable. and since the utility creates the /var/lib/firmware/bitstream automatically I do not think a systemd operation is strictly neccessary.